gcm_filters/filter.py

```diff
@@ -191,6 +191,53 @@ def filter_func(field, *args):
     return filter_func
 
 
+def _create_filter_func_vec(
+    filter_spec: FilterSpec,
+    Laplacian: BaseLaplacian,
+):
+    """Returns a function whose first two arguments are the vector components of the field to be filtered
+    and whose subsequent arguments are the require grid variables
+    """
+
+    def filter_func_vec(ufield, vfield, *args):
+        # these next steps are a kind of hack we have to turn keyword arugments into regular arguments
+        # the reason for doing this is that Xarray's apply_ufunc machinery works a lot better
+        # with regular arguments
+        assert len(args) == len(Laplacian.required_grid_args())
+        grid_vars = {k: v for k, v in zip(Laplacian.required_grid_args(), args)}
+        laplacian = Laplacian(**grid_vars)
```

Comment:
What happens if the user passes a non-vector Laplacian here? It should not be allowed.